### PR TITLE
`CupertinoSliverNavigationBar`: Add example

### PR DESCRIPTION
**Pull request by TahaTesser**
Part of https://github.com/flutter/flutter/issues/72926

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      // A ScrollView that creates custom scroll effects using slivers.
+      child: CustomScrollView(
+        // A list of sliver widgets.
+        slivers: <Widget>[
+          const CupertinoSliverNavigationBar(
+            leading: Icon(CupertinoIcons.person_2),
+            // This title is visible in both collapsed and expanded states.
+            // when `middle` parameter is omitted, it takes the `largeTitle`.
```
Comment (gspencergoog) [inline examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart:38]:
```suggestion
            // When the "middle" parameter is omitted, it takes the "largeTitle".
```

And what do you mean by "takes"?  Uses?

File: examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            largeTitle: Text('Contacts'),
+            trailing: Icon(CupertinoIcons.add_circled),
+          ),
+          // This widget fills the remaining space in the viewport.
+          // Drag the scrollable area to collapse CupertinoSliverNavigationBar.
```
Comment (gspencergoog) [inline examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart:43]:
```suggestion
          // Drag the scrollable area to collapse the CupertinoSliverNavigationBar.
```

File: examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            border: Border(
+              bottom: BorderSide(
+                color: brightness == Brightness.light
+                  ? CupertinoColors.black
+                  : CupertinoColors.white,
+              ),
+            ),
+            // The middle widget is visible in both collapsed and expanded states.
+            middle: const Text('Contacts Group'),
+            // When `middle` parameter is implemented, larget title is only visible
```
Comment (gspencergoog) [inline examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart:86]:
```suggestion
            // When the "middle" parameter is implemented, the largest title is only visible
```

File: examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            // when the CupertinoSliverNavigationBar is fully expanded.
+            largeTitle: const Text('Family'),
+          ),
+          SliverFillRemaining(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: const <Widget>[
+                Text('Drag me up', textAlign: TextAlign.center),
+                // When `leading` parameter is omitted on a route that has a previous page,
```
Comment (gspencergoog) [inline examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart:95]:
```suggestion
                // When the "leading" parameter is omitted on a route that has a previous page,
```

These aren't Dartdoc comments, so use regular quotes instead.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Just some nits.  Thanks for the example!

File: examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart (hunk 1 above)
Comment (TahaTesser) [reply examples/api/lib/cupertino/nav_bar/cupertino_sliver_nav_bar.0.dart]: Yes

I  just updated this description like this to make it more clear:
```console
            // This title is visible in both collapsed and expanded states.
            // When the "middle" parameter is omitted, the widget provided
            // in the "largeTitle" parameter is used instead in the collapsed state.
```

**Pull request merged**